File: Evergrande/src/gestorAplicacion/otros/UnidadCerrada.java
package gestorAplicacion.otros;
import java.io.Serializable;

public class UnidadCerrada extends UnidadResidencial{
	private static final long serialVersionUID = 1L;
	private int numero_torres;
	
	public UnidadCerrada(int idUnidadResidencial, String nombre, String barrio, int numero_torres) {
		super(idUnidadResidencial, nombre, barrio);
		this.numero_torres = numero_torres;
	}
	
	public String presentarInforme() {
		return  " Nombre de la Unidad: " + this.getNombre() + "\n" + 
				" Identificada con el id: " + this.getIdUnidadResidencial() + "\n" + 
				" Ubicada en el barrio " + this.getBarrio() + " en la ciudad de Medellin, cuyo alcalde es "+ ALCALDE + " y actualmente presenta un IVA del "+ IVA + "\n" +
				" Con " + this.getNumero_torres() + " torres" + "\n";
	}
	
    public int getNumero_torres() {
        return this.numero_torres;
    }

    public void setNumero_torres(int numero_torres) {
        this.numero_torres = numero_torres;
    }

}

File: Python/src/gestorAplicacion/otros/UnidadCerrada.py
'''
Autores: 
Julián Orozco Vanegas
Juan Nicolas Piedrahita Salas
David Escobar Ruiz

Esta clase simula el comportamiento de una unidad residencial de tipo cerrada y sus caracteristicas
'''

from gestorAplicacion.otros.UnidadResidencial import UnidadResidencial

class UnidadCerrada(UnidadResidencial):
  def __init__(self, idUnidadResidencial, nombre, barrio, numero_torres):
    super().__init__(idUnidadResidencial, nombre, barrio)
    self.numero_torres = numero_torres

  def presentarInforme(self):
    return "UNIDAD CERRADA\n" + " Nombre de la Unidad: " + super().getNombre() + "\n" + " Identificada con el id: " + str(super().getIdUnidadResidencial()) + "\n" + " Ubicada en el barrio " + super().getBarrio() + " en la ciudad de Medellin, cuyo alcalde es "+ super().ALCALDE + " y actualmente presenta un IVA del "+ str(super().IVA) + "\n" + " Con " + str(self.getNumero_torres()) + " torres" + "\n"
  
  def getNumero_torres(self):
    return self.numero_torres
  
  def setNumero_torres(self, nNumero_torres):
    self.numero_torres = nNumero_torres